### PR TITLE
Write queuing model section

### PR DESCRIPTION
**Pull request by daffidwilde**
A section to summarise how the queuing model is to be constructed. This revolves around simulating a multi-class M|M|c queue with a series of parameters and comparing the records to those in the dataset from Cwm Taf.

### CODE REVIEW

File: sections/model.tex [hunk 1]
```diff
@@ -0,0 +1,139 @@
+\section{Constructing the queuing model}\label{sec:model}
+
+% TODO What exactly is the problem and how can it be addressed?
+
+Owing to a lack of available data on the system and its patients, the options
+for the queuing model used are limited compared to those employed in some modern
+works. However, there is a precedent for simplifying healthcare systems to a
+single node with, for example, parallel servers that emulate resource
+availability.~\cite{Steins2013}~and~\cite{Williams2015} provide a good examples
```
Comment (drvinceknight) [inline sections/model.tex]:
```suggestion
availability.~\cite{Steins2013}~and~\cite{Williams2015} provide good examples
```

File: sections/model.tex [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+of how this approach, when paired with discrete event simulation, can expose the
+resource needs of a system beyond deterministic queuing theory models. In
+particular,~\citeauthor{Williams2015} show how a single node, multiple server
```
Comment (drvinceknight) [inline sections/model.tex]:
```suggestion
particular,~\citeauthor{Williams2015} shows how a single node, multiple server
```

File: sections/model.tex [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+queue is employed to model a hypothetical ward of patients presenting COPD.\ In
+addition to this, the grouping found in Section~\ref{subsec:overview} provides
+a set of patient classes in the queue. Under this model, the following
+assumptions are made:
+\begin{enumerate}
+    \item Inter-arrival and service times of patients are each exponentially
+        distributed with some mean. This is in spite of the system time
+        distributions shown in Figure~\ref{fig:clusters_los} in order to
+        simplify the model parameterisation.
+    \item There are $c$ servers available to arriving patients at the node
```
Comment (drvinceknight) [inline sections/model.tex]:
```suggestion
    \item There are \(c\) servers available to arriving patients at the node
```

Comment (drvinceknight) [reply sections/model.tex]: :)

File: sections/model.tex [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+As such, a reasonable candidate is the (first) Wasserstein metric, also known as
+the `earth mover' or `digger' distance~\cite{Vaserstein1969}. The Wasserstein
+metric satisfies the conditions of a formal mathematical metric (like the
+typical Euclidean distance), and its values take the units of the distributions
+under comparison (in this case: days). Both of these characteristics can aid
+understanding and explanation. In simple terms, the distance measures the
+approximate `minimal work' required to move between two probability
+distributions where `work' can be loosely defined as the product of how much of
+the distribution's mass is to moved and the distance it must be moved
```
Comment (drvinceknight) [inline sections/model.tex]:
```suggestion
the distribution's mass is to be moved and the distance it must be moved
```

File: sections/model.tex [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+\noindent where \(F\) and \(G\) are the cumulative density functions of \(U\)
+and \(V\) respectively. A proof of~\eqref{eq:wasserstein} is presented
+in~\cite{Ramdas2017}. The parameter set with the smallest maximum distance
+between any cluster's simulated system time distribution and the overall
+observed length of stay distribution is then taken to be the most appropriate.
+
+Specifically, the parameter sweep included values of each \(p_i\) from \(0.5\)
+to \(1\) with a granularity of \(10^{-1}\) and values of \(c\) from \(40\) to
+\(60\) at steps of \(5\). These choices were informed by the assumptions of the
+model and formative analysis to reduce the parameter space given the
```
Comment (drvinceknight) [inline sections/model.tex]:
```suggestion
Specifically, the parameter sweep included values of each \(p_i\) from \(0.5\)
to \(1\) with a granularity of \(10^{-1}\) and values of \(c\) from \(40\) to
\(60\) at steps of \(5\).  % TODO Update these values when the new results come back
These choices were informed by the assumptions of the
model and formative analysis to reduce the parameter space given the
```

File: sections/model.tex [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+Figure~\ref{fig:worst_params} highlights the importance of good parameter
+estimation under this model since the likelihood of short-stay patient arrivals
+has been inflated disproportionately against the tail of the distribution.
+Table~\ref{tab:comparison} reinforces these results numerically, showing a
+clear fit by the best parameters across the board.
+
+\begin{table}
+    \centering
+    \input{tex/comparison}
+    \caption{A comparison of summary statistics between the observed data, and
```
Comment (drvinceknight) [inline sections/model.tex]: Can we include the Wasserstein distance in the table? (It would be 0 for the observed one).

File: sections/model.tex [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+In order to evaluate appropriate values of each \(p_i\) and the value of \(c\),
+the system is simulated across a parameter space. The objective then is to
+determine which set of parameters provides the most realistic analogue of the
+observed data. Given that the length of stay is one of the few ground truths
+available in the provided data, and given that length of stay and resource
+availability are connected, the output of this model is the simulated length of
+stay. With this simulated data, a parameter set can be evaluated by comparing
+its associated system time distribution with the real length of stay
+distribution.
```
Comment (drvinceknight) [inline sections/model.tex]: I can't exactly pin point what I don't like about this paragraph. Would something like the following work:

```suggestion
One of the few ground truths available in the provided data is the distribution 
of the total length of stay. Given that the length of stay and resource availability 
are connected, the approach here will be to simulate the length of stay distribution 
for a range of values \(p_i\) and \(c\) in order to find the parameters that best 
match the observed data.
```

File: sections/model.tex (hunk 6 above)
Comment (drvinceknight) [reply sections/model.tex]: Can we also include the c and the p?

File: sections/model.tex [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
```
Comment (drvinceknight) [inline sections/model.tex:61]: Can we include one of your nice diagrams here? (Perhaps something for us to chat about.)

File: sections/model.tex [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+The results of this parameter sweep can be summarised in
+Figures~\ref{fig:best_params}~and~\ref{fig:worst_params}. Each figure shows a
+comparison of the observed lengths of stay across all groups and the newly
+simulated data with the best and worst parameter sets respectively. It can be
+seen that, in the best case, a very close fit has been found. Meanwhile,
+Figure~\ref{fig:worst_params} highlights the importance of good parameter
+estimation under this model since the likelihood of short-stay patient arrivals
+has been inflated disproportionately against the tail of the distribution.
+Table~\ref{tab:comparison} reinforces these results numerically, showing a
+clear fit by the best parameters across the board.
```
Comment (drvinceknight) [inline sections/model.tex:138]: Can we discuss the best/worst fit parameters and what they mean here.

**Review by drvinceknight — CHANGES_REQUESTED** [review]
Looking good @daffidwilde, some suggestions.

File: sections/model.tex (hunk 6 above)
Comment (daffidwilde) [reply sections/model.tex]: That would be sensible. I've included them in 6824457.

File: sections/model.tex (hunk 2 above)
Comment (daffidwilde) [reply sections/model.tex]: Fixed in fbf8350

File: sections/model.tex (hunk 9 above)
Comment (daffidwilde) [reply sections/model.tex:138]: Added supporting definition in cf45d0b

File: sections/model.tex (hunk 8 above)
Comment (daffidwilde) [reply sections/model.tex:61]: I'm going to leave that out of this PR but I'll open an issue.

**Pull request merged**